### PR TITLE
Adding loading check, edit functionality and 404 page

### PR DESCRIPTION
**Pull request by AbdulBasit0044**
In this commit these things are implemented:

-Edit movies functionality
-Loading check
-Shows error
-404 Page

### CODE REVIEW

File: src/components/movie/MovieDetails.jsx [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+import { MovieSharp } from "@mui/icons-material";
 
 const MovieDetails = () => {
   const { id } = useParams();
   const navigate = useNavigate();
-  const { movies: movie, isPending, error } = GetMoviesBasedOnId(id);
+  const { movies: movie, error } = GetMoviesBasedOnId(id);
+
+  const handleEdit = () => {
+  };
```
Comment (omerzn) [inline src/components/movie/MovieDetails.jsx]: where is edit functionality?

Comment (omerzn) [reply src/components/movie/MovieDetails.jsx]: Please also a one line space between functions

File: src/components/navbar/Edit.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  const navigate = useNavigate();
+  const { movies } = GetMoviesBasedOnId(id);
+  const [movie, setMovie] = useState(null);
+  const [title, setTitle] = useState("");
+  const [director, setDirector] = useState("");
+  const [overview, setOverview] = useState("");
+  const [rating, setRating] = useState("");
+  const [img, setImg] = useState();
+
+
```
Comment (omerzn) [inline src/components/navbar/Edit.jsx]: extra space

File: src/components/navbar/Edit.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  useEffect(() => {
+    if (movies != null) {
+      setTitle(movies.title);
+      setDirector(movies.director);
+      setOverview(movies.overview);
+      setRating(movies.rating);
+      setImg(movies.img);
+    }
+  }, [movies]);
```
Comment (omerzn) [inline src/components/navbar/Edit.jsx]: add space after this

File: src/components/navbar/Edit.jsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  const [movie, setMovie] = useState(null);
+  const [title, setTitle] = useState("");
+  const [director, setDirector] = useState("");
+  const [overview, setOverview] = useState("");
+  const [rating, setRating] = useState("");
+  const [img, setImg] = useState();
+
+
+  useEffect(() => {
+    if (movies != null) {
```
Comment (omerzn) [inline src/components/navbar/Edit.jsx]: let's discuss this

File: src/hooks/useFetch.jsx [hunk 5]
```diff
@@ -2,22 +2,29 @@ import { useState, useEffect } from "react";
 
 const useFetch = (url) => {
   const [movies, setMovies] = useState(null);
-  const [isPending, setIsPending] = useState(null);
   const [error, setError] = useState(null);
 
   useEffect(() => {
-    fetch(url)
-      .then((resp) => {
+    const abortCont = new AbortController();
```
Comment (omerzn) [inline src/hooks/useFetch.jsx]: why did you use this?

**Review by omerzn — CHANGES_REQUESTED** [review]

File: src/components/movie/MovieDetails.jsx (hunk 1 above)
Comment (AbdulBasit0044) [reply src/components/movie/MovieDetails.jsx]: Edit functionality is implemented in the Edit component, I forgot to remove this function from here, now this is updated.

File: src/components/navbar/Edit.jsx (hunk 2 above)
Comment (AbdulBasit0044) [reply src/components/navbar/Edit.jsx]: Fixed

File: src/components/navbar/Edit.jsx (hunk 4 above)
Comment (AbdulBasit0044) [reply src/components/navbar/Edit.jsx]: Okay

**Review by omerzn — APPROVED** [review]

**Pull request merged**